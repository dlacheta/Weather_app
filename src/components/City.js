import React from 'react';
import PropTypes from 'prop-types';

const City = ({
  city: {
    name,
    country,
    population,
    coord: { lat, lon },
  },
}) => (
  <div className="card">
    <div className="card-body">
      <h4 className="card-title">{`${name}, ${country}`}</h4>

      <div className="d-flex justify-content-between">
        <p>{`Population: ${population}`}</p>
        <p>{`Coordinates: Latitute: ${lat} Longitude: ${lon}`}</p>
      </div>
    </div>
  </div>
);

City.propTypes = {
  city: PropTypes.shape({
    name: PropTypes.string,
    country: PropTypes.string,
    population: PropTypes.number,
    coord: PropTypes.shape({
      lat: PropTypes.number,
      lon: PropTypes.number,
    }),
  }).isRequired,
};

export default City;
