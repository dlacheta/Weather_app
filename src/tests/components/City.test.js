import React from 'react';
import { shallow } from 'enzyme';
import City from '../../components/City';

it('City component renders correctly', () => {
  const city = {
    name: 'London',
    country: 'GB',
    population: 1000000,
    coord: {
      lat: 51.073,
      lon: -0.1277,
    },
  };
  const wrapper = shallow(<City city={city} />);
  expect(wrapper).toMatchSnapshot();
});
