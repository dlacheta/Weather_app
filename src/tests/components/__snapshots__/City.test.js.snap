// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <City
    city={
      Object {
        "coord": Object {
          "lat": 51.073,
          "lon": -0.1277,
        },
        "country": "GB",
        "name": "London",
        "population": 1000000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-body"
      >
        <h4
          className="card-title"
        >
          London, GB
        </h4>
        <div
          className="d-flex justify-content-between"
        >
          <p>
            Population: 1000000
          </p>
          <p>
            Coordinates: Latitute: 51.073 Longitude: -0.1277
          </p>
        </div>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="card-title"
          >
            London, GB
          </h4>,
          <div
            className="d-flex justify-content-between"
          >
            <p>
              Population: 1000000
            </p>
            <p>
              Coordinates: Latitute: 51.073 Longitude: -0.1277
            </p>
          </div>,
        ],
        "className": "card-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "London, GB",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "London, GB",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Population: 1000000
              </p>,
              <p>
                Coordinates: Latitute: 51.073 Longitude: -0.1277
              </p>,
            ],
            "className": "d-flex justify-content-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population: 1000000",
              },
              "ref": null,
              "rendered": "Population: 1000000",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coordinates: Latitute: 51.073 Longitude: -0.1277",
              },
              "ref": null,
              "rendered": "Coordinates: Latitute: 51.073 Longitude: -0.1277",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-body"
        >
          <h4
            className="card-title"
          >
            London, GB
          </h4>
          <div
            className="d-flex justify-content-between"
          >
            <p>
              Population: 1000000
            </p>
            <p>
              Coordinates: Latitute: 51.073 Longitude: -0.1277
            </p>
          </div>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="card-title"
            >
              London, GB
            </h4>,
            <div
              className="d-flex justify-content-between"
            >
              <p>
                Population: 1000000
              </p>
              <p>
                Coordinates: Latitute: 51.073 Longitude: -0.1277
              </p>
            </div>,
          ],
          "className": "card-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "London, GB",
              "className": "card-title",
            },
            "ref": null,
            "rendered": "London, GB",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Population: 1000000
                </p>,
                <p>
                  Coordinates: Latitute: 51.073 Longitude: -0.1277
                </p>,
              ],
              "className": "d-flex justify-content-between",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Population: 1000000",
                },
                "ref": null,
                "rendered": "Population: 1000000",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Coordinates: Latitute: 51.073 Longitude: -0.1277",
                },
                "ref": null,
                "rendered": "Coordinates: Latitute: 51.073 Longitude: -0.1277",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
